### DISCUSSION TITLE
Segmentation Fault when running input file from terminal

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#18385] [Q&A General] [answered]
Hi all, 

I'm getting the following segmentation fault error, I checked with gdb and this is what I got:

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7ef9ff7 in xxxMaterial::validParams ()

Does it have to with my code, or is it something else? I can't seem to figure it out, as the fault is occurring before anything is even done (line 8). All my required material properties are initialized in the input file, so I don't have any idea what else could be causing this. I'm using Ubuntu 20.04 on Windows (WSL).

**Comment by GiudGiud** [comment, marked as answer]
hello 

If you using gdb could you please provide us with the backtrace then?
If you could build in debug mode (METHOD=dbg make) first, that would help since it ll give the exact line of the set fault. 

Alternatively, you could run valgrind and it ll point out where the memory error happened

 Best
Guillaume

**Reply by richmondodufisan** [reply]
Thanks, I was able to figure it out with the backtrace, I didn't know about that before. Also, a quick question: I'm automatically calculating the Jacobian instead of doing it manually, because it's a system of 4 coupled PDEs. It converged and it's pretty convenient, but I was wondering why I'm getting this output on the terminal

![output](https://user-images.githubusercontent.com/52802063/126255122-5ed7dc4e-6d97-4a5e-975b-08b7cf642b20.JPG)

I usually get "neatly arranged" tables while it's solving, but I guess it's fine since I still get a good result. This just threw me off at first.

**Reply by GiudGiud** [reply]
This looks like you have a print statement somewhere. Like in a kernel or a material property. And it s flooding the console output

**Reply by WilkAndy** [reply]
Look for "std::cout", "std::cerr" lines

**Comment by WilkAndy** [comment]
I'd hazard a guess that it's your "xxxMaterial": look in the validParams() function.
